v2/pkg/astnormalization/astnormalization.go

```diff
@@ -193,7 +190,21 @@ func WithNormalizeDefinition() Option {
 }
 
 func (o *OperationNormalizer) setupOperationWalkers() {
-	o.operationWalkers = make([]walkerStage, 0, 6)
+	o.operationWalkers = make([]walkerStage, 0, 9)
+
+	// NOTE: normalization rules for variables relies on the fact that
+	// we will visit only single operation, so it is important to remove non-matching operations
+	if o.options.removeNotMatchingOperationDefinitions {
+		removeNotMatchingOperationDefinitionsWalker := astvisitor.NewWalker(2)
```

Comment:
Alternatively, you could also just "look" into the AST directly without walking.